overlord/overlord.go

```diff
@@ -117,12 +118,31 @@ func (o *Overlord) ensureTimerSetup() {
 	o.ensureNext = time.Now().Add(ensureInterval)
 }
 
-func (o *Overlord) ensureTimerReset() {
+func (o *Overlord) syncEnsurePrep() time.Time {
 	o.ensureLock.Lock()
 	defer o.ensureLock.Unlock()
 	now := time.Now()
 	o.ensureTimer.Reset(ensureInterval)
 	o.ensureNext = now.Add(ensureInterval)
+	o.ensureRunning++
```

Comment:
It's not running.. haven't read it all, but this probably has a different meaning.